admin/app.py

```diff
@@ -50,6 +51,9 @@ def page_not_found(e):
         if user_search:
             user_search = user_search[0]
             url_to_redirect += f'&{"userSearch"}={user_search}'
+        if book_search:
+            book_search = book_search[0]
```

Comment:
It is a bad approach to redefine variable value and change its type. In the current case from list to str.
The same with user_search. 
I guess that we can use 
user_search_result =  user_search[0]
book_search_result = book_search[0]